src/cryptonote_core/tx_pool.cpp

```diff
@@ -515,6 +515,59 @@ namespace cryptonote
     return true;
   }
   //---------------------------------------------------------------------------------
+  bool tx_memory_pool::get_transaction_info(const crypto::hash &txid, tx_details &td) const
+  {
+    PERF_TIMER(get_transaction_info);
+    CRITICAL_REGION_LOCAL(m_transactions_lock);
+    CRITICAL_REGION_LOCAL1(m_blockchain);
+
+    try
+    {
+      LockedTXN lock(m_blockchain);
+      txpool_tx_meta_t meta;
+      if (!m_blockchain.get_txpool_tx_meta(txid, meta))
+      {
+        MERROR("Failed to find tx in txpool");
+        return false;
+      }
+      cryptonote::blobdata txblob = m_blockchain.get_txpool_tx_blob(txid);
+      auto ci = m_parsed_tx_cache.find(txid);
+      if (ci != m_parsed_tx_cache.end())
+      {
+        td.tx = ci->second;
+      }
+      else if (!(meta.pruned ? parse_and_validate_tx_base_from_blob(txblob, td.tx) : parse_and_validate_tx_from_blob(txblob, td.tx)))
```

Comment:
I believe `meta.pruned` doesn't exist untill you merge #5915 to `master`

Reply:
Good catch, thanks. If it doesn't get merged I'll remove it then.